packages/twenty-server/src/database/commands/upgrade-version-command/1-5/1-5-populate-message-folder-fields.command.ts

```diff
@@ -0,0 +1,322 @@
+  constructor(
+    @InjectRepository(Workspace)
+    protected readonly workspaceRepository: Repository<Workspace>,
+    protected readonly twentyORMGlobalManager: TwentyORMGlobalManager,
+    protected readonly workspaceSchemaManager: WorkspaceSchemaManagerService,
+    private readonly syncMessageFoldersService: SyncMessageFoldersService,
+
+    @InjectDataSource()
+    protected readonly coreDataSource: DataSource,
+    private readonly workspaceMetadataVersionService: WorkspaceMetadataVersionService,
+  ) {
+    super(workspaceRepository, twentyORMGlobalManager);
+  }
+
+  override async runOnWorkspace({
+    index,
+    total,
+    workspaceId,
+  }: RunOnWorkspaceArgs): Promise<void> {
+    this.logger.log(
+      `Populating message folder fields for workspace ${workspaceId} ${index + 1}/${total}`,
+    );
+
+    try {
+      const basicUpdateResult =
+        await this.populateExternalIdsForHardcodedFolders(workspaceId);
+
+      if (basicUpdateResult.failed.length > 0) {
+        this.logger.warn(
+          `Workspace ${workspaceId}: Basic update - Processed ${basicUpdateResult.processed}, Updated ${basicUpdateResult.updated}, Failed: ${basicUpdateResult.failed.length}`,
+        );
+      }
+
+      this.logger.log(
+        `Workspace ${workspaceId}: Basic update - Processed ${basicUpdateResult.processed}, Updated ${basicUpdateResult.updated} message folders`,
+      );
+
+      const syncResult = await this.syncAllMessageChannelFolders(workspaceId);
+
+      this.logger.log(
+        `Workspace ${workspaceId}: Synced folders for ${syncResult.channelsSynced} message channels`,
+      );
+    } catch (error) {
+      this.logger.error(
+        `Failed to populate message folder fields for workspace ${workspaceId}: ${error.message}`,
+      );
+
+      throw error;
+    }
+  }
+
+  private async populateExternalIdsForHardcodedFolders(
+    workspaceId: string,
+  ): Promise<{
+    processed: number;
+    updated: number;
+    failed: Array<{ id: string; name: string; error: string }>;
+  }> {
+    const folders = await this.getFoldersNeedingMigration(workspaceId);
+
+    if (folders.length === 0) {
+      this.logger.log(
+        `No message folders need updating in workspace ${workspaceId}`,
+      );
+
+      return { processed: 0, updated: 0, failed: [] };
+    }
+
+    this.logger.log(
+      `Found ${folders.length} message folders to update in workspace ${workspaceId}`,
+    );
+
+    const { channelMap, channelExternalFoldersMap } =
+      await this.buildChannelDataMaps(folders, workspaceId);
+
+    return await this.updateFolderExternalIds(
+      folders,
+      channelMap,
+      channelExternalFoldersMap,
+      workspaceId,
+    );
+  }
+
+  private async syncAllMessageChannelFolders(
+    workspaceId: string,
+  ): Promise<{ channelsSynced: number }> {
+    const messageChannelRepository =
+      await this.twentyORMGlobalManager.getRepositoryForWorkspace<MessageChannelWorkspaceEntity>(
```

Comment:
what you are doing is fine but so you know for next time: RunOnWorkspaceArgs is already providing an instanciated workspaceDatasource